python/lsst/daf/butler/registry/interfaces/_database.py

```diff
@@ -1875,6 +1879,25 @@ def query(
             if connection is not self._session_connection:
                 connection.close()
 
+    def _log_query_and_plan(
+        self,
+        connection: sqlalchemy.Connection,
+        sql: sqlalchemy.sql.expression.Executable | sqlalchemy.sql.expression.SelectBase,
+    ) -> None:
+        """Log the given SQL statement and the DB's plan for executing it if
+        the environment variable DAF_BUTLER_DEBUG_QUERIES is set to a truthy
+        value.
+        """
+        if os.environ.get("DAF_BUTLER_DEBUG_QUERIES", False):
+            assert isinstance(sql, sqlalchemy.SelectBase)
+            compiled = sql.compile(connection)
+            print(
+                f"Executing SQL statement:\n{compiled}\nBind parameters: {compiled.params}", file=sys.stderr
+            )
+
+            query_plan = get_query_plan(connection, sql)
+            print(f"Query plan:\n{query_plan}", file=sys.stderr)
```

Comment:
Maybe send this to `logging`?

Reply:
Yeah I considered that but then to use it you'd have to also figure out how to get logging turned on, so it seemed easier to just send it to stderr.  Not opposed to it if you want to make that change yourself though.